vita3k/gxm/src/textures.cpp

```diff
@@ -17,47 +17,280 @@
 
 #include <gxm/functions.h>
 #include <map>
+#include <util/align.h>
+
+#include <assert.h>
 
 namespace gxm {
-size_t get_width(const SceGxmTexture *texture) {
-    if ((texture->type << 29 != SCE_GXM_TEXTURE_SWIZZLED) && (texture->type << 29 != SCE_GXM_TEXTURE_CUBE)) {
-        return texture->width + 1;
+uint32_t get_width(const SceGxmTexture &texture) {
+    if (texture.texture_type() != SCE_GXM_TEXTURE_SWIZZLED && texture.texture_type() != SCE_GXM_TEXTURE_CUBE) {
+        return texture.width + 1;
     }
-    return 1ull << (texture->width_base2 & 0xF);
+    return 1u << texture.width_base2;
 }
 
-size_t get_height(const SceGxmTexture *texture) {
-    if ((texture->type << 29 != SCE_GXM_TEXTURE_SWIZZLED) && (texture->type << 29 != SCE_GXM_TEXTURE_CUBE)) {
-        return texture->height + 1;
+uint32_t get_height(const SceGxmTexture &texture) {
+    if (texture.texture_type() != SCE_GXM_TEXTURE_SWIZZLED && texture.texture_type() != SCE_GXM_TEXTURE_CUBE) {
+        return texture.height + 1;
     }
-    return 1ull << (texture->height_base2 & 0xF);
+    return 1ull << texture.height_base2;
```

Comment:
```suggestion
    return 1u << texture.height_base2;
```
to the same code as in ` get_width`